apps/sms/js/thread_list_ui.js

```diff
@@ -449,75 +457,27 @@ var ThreadListUI = {
   // please make sure url will also be revoked if new delete api remove threads
   // without calling removeThread in the future.
   delete: function thlui_delete(selected) {
-    function performDeletion() {
-    /* jshint validthis: true */
-
-      var threadIdsToDelete = [],
-          messageIdsToDelete = [],
-          threadCountToDelete = 0;
-
-      function exitEditMode() {
-        ThreadListUI.cancelEdit();
-        WaitingScreen.hide();
-      }
-
-      function onAllThreadMessagesRetrieved() {
-        if (!--threadCountToDelete) {
-          MessageManager.deleteMessages(messageIdsToDelete);
-
-          threadIdsToDelete.forEach(function(threadId) {
-            ThreadListUI.deleteThread(threadId);
-          });
-
-          messageIdsToDelete = threadIdsToDelete = null;
-
-          exitEditMode();
-        }
-      }
-
-      function onThreadMessageRetrieved(message) {
-        messageIdsToDelete.push(message.id);
-        return true;
-      }
+    function performDeletion(selected) {
+      /* jshint validthis: true */
+      var threadIds = [], draftIds = [];
 
       WaitingScreen.show();
 
-      threadIdsToDelete = selected.reduce(function(list, value) {
+      threadIds = selected.reduce(function(list, value) {
         // Coerce the threadId back to a number MobileMessageFilter and all
         // other platform APIs expect this value to be a number.
         var threadId = +value;
         var isDraft = typeof Threads.get(threadId) === 'undefined';
 
         if (isDraft) {
-          Drafts.delete(Drafts.get(threadId));
-          ThreadListUI.removeThread(threadId);
+          draftIds.push(threadId);
         } else {
           list.push(threadId);
         }
-
         return list;
       }, []);
```

Comment:
Yeah That's also fine. But then,  there is no need of delete function.. right? we will have 'deleteThreadDraftUI ' instead. 
I didn't remove the delete method (and that loop) because of one reason.. delete method is used at many places (and test too). So i didn't modify the name of 'delete' method name.